pkg/featureflags/featureflags_test.go

```
@@ -0,0 +1,376 @@
+
+func TestEmbeddedProvider_Init_ValidFile(t *testing.T) {
+	p := newProvider(t)
+	initProvider(t, p)
+}
+
+func TestEmbeddedProvider_Init_MissingFile(t *testing.T) {
+	p, err := NewEmbeddedProvider(commoncfg.FeatureFlags{FilePath: "/nonexistent/flags.yaml"})
+	if err != nil {
+		t.Fatalf("unexpected construction error: %v", err)
+	}
+	sh := p.(openfeature.StateHandler)
```

golangci-lint (2.12.2)

[error] type assertion must be checked

(forcetypeassert)